### PR TITLE
[google_sign_in] Updates google_sign_in_platform_interfaces adding parametrized `clientId`

### PR DESCRIPTION
**Pull request by ggirotto**
This PR adds support to https://github.com/flutter/plugins/pull/3640 by parametrizing `clientId` property in `MethodChannelGoogleSignIn`

*List which issues are fixed by this PR. You must list at least one issue.*

Same issues as https://github.com/flutter/plugins/pull/3640

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by ggirotto** [comment]
Testes to `google_sign_in` were added at https://github.com/flutter/plugins/pull/3640, which will complement the current PR changes

**Comment by cyanglaz** [comment]
platform interface change should also be testable through unit tests.

**Comment by ggirotto** [comment]
@cyanglaz Updated `Other functions pass through arguments to the channel` test which covers parametrized arguments in `init` function

File: packages/google_sign_in/google_sign_in_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 2.0.1
```
Comment (cyanglaz) [inline packages/google_sign_in/google_sign_in_platform_interface/CHANGELOG.md:1]:
```suggestion
## 2.0.1

```

**Review by cyanglaz — APPROVED** [review]
LGTM! Thanks

File: packages/google_sign_in/google_sign_in_platform_interface/pubspec.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ description: A common platform interface for the google_sign_in plugin.
 homepage: https://github.com/flutter/plugins/tree/master/packages/google_sign_in/google_sign_in_platform_interface
 # NOTE: We strongly prefer non-breaking changes, even at the expense of a
 # less-clean API. See https://flutter.dev/go/platform-interface-breaking-changes
-version: 2.0.0
+version: 2.0.1
```
Comment (cyanglaz) [inline packages/google_sign_in/google_sign_in_platform_interface/pubspec.yaml:6]: 2.1.0

**Review by cyanglaz — CHANGES_REQUESTED** [review]
Need to update the minor version because there's a public api change.

**Comment by ggirotto** [comment]
@cyanglaz done!

File: packages/google_sign_in/google_sign_in_platform_interface/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+## 2.1.0
```
Comment (cyanglaz) [inline packages/google_sign_in/google_sign_in_platform_interface/CHANGELOG.md:1]: need extra line after version :) 
```suggestion
## 2.1.0

```

**Review by cyanglaz — COMMENTED** [review]
Actually im sorry. The clientId field was an existing unused filed in the public API, so your initial version was correct
ill change it back

**Review by cyanglaz — COMMENTED** [review]
Rebasing master should fix ci

**Comment by ggirotto** [comment]
@cyanglaz Now I think its good. Please let me know if it need more adjustments

**Review by cyanglaz — APPROVED** [review]
LGTM

**Pull request merged**